### DISCUSSION TITLE
using (var file = new StreamWriter(path))

### DISCUSSION THREAD

**Discussion by nicolasvagnes** [balta-io/2801#98] [Perguntas e Respostas] [answered]
Utilizei o using System.IO;

Porém da duas mensagens de erro:

PS C:\dev\TextEditor> dotnet run
C:\dev\TextEditor\Program.cs(59,28): error CS0103: O nome "text" não existe no contexto atual [C:\dev\TextEditor\TextEditor.csproj]

Ocorreu uma falha no build. Corrija os erros de build e execute novamente.

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @nicolasvagnes, como vai?

Consegue colar o código do seu programa aqui? Ou um print? Só com esta mensagem não consigo dizer o que pode ser o erro.

**Reply by nicolasvagnes** [reply]
Vou bem e você?
Segue o código: 

using System;
using System.IO;

namespace TextEditor
{
    class Program
    {
        static void Main(string[] args)
        {
            Menu();
        }  
        static void Menu()
        {
            Console.Clear();
            Console.WriteLine("What do you want?");
            Console.WriteLine("1- Open file");
            Console.WriteLine("2- Create new file");
            Console.WriteLine("0- Exit");
            short option = short.Parse(Console.ReadLine());

            switch (option)
            {
                case 0: System.Environment.Exit(0); break;
                case 1: Open(); break;
                case 2: Edit(); break;
                default: Menu(); break;
            }
        }
        static void Open()
        {
        }
        static void Edit()
        {
            Console.Clear();
            Console.WriteLine("Type your text bellow: (ESC to close)");
            Console.WriteLine("_____________________________________");
            string text = "";

            do {
                text += Console.ReadLine();
                text += Environment.NewLine;
            }
            while(Console.ReadKey().Key != ConsoleKey.Escape);

            Console.Write(text);
        }
        static void Save()
        {
            Console.Clear();
            Console.WriteLine("What way to save the file?");
            var path = Console.ReadLine();

            using (var file = new StreamWriter(path))
            {
                file.Write(text);
            }
        }
    }
}

**Comment by andrebaltieri** [comment]
Bom dia, @nicolasvagnes como vai?

No caso, você não declarou a variável `text` em nenhum lugar... veja seu método edit por exemplo:
```csharp
static void Edit()
{
	Console.Clear();
	Console.WriteLine("Type your text bellow: (ESC to close)");
	Console.WriteLine("_____________________________________");
	string text = "";
	do
	{
		text += Console.ReadLine();
		text += Environment.NewLine;
	}
	while (Console.ReadKey().Key != ConsoleKey.Escape);
	Console.Write(text);
}
```

Já seu método `Save()` não tem a variável `text` declarada...
```sharp
static void Save()
{
	Console.Clear();
	Console.WriteLine("What way to save the file?");
	var path = Console.ReadLine();
	using (var file = new StreamWriter(path))
	{
		file.Write(text);
	}
}
```

Desta forma o comando `file.Write(text);` vai dar erro pois a variável `text` não existe...
Agora precisa entender o que você gostaria de fazer... se é ler algum input do usuário neste momento, se é receber um texto via parâmetro...
